openvdb/openvdb/python/setup.py

```diff
@@ -0,0 +1,44 @@
+import sys
+import pathlib
+import numpy as np
+from skbuild import setup
+
+
+with open(pathlib.Path(__file__).parent / "README.md") as f:
+    long_description = f.read()
+
+
+# Explicitly set NUMPY_INCLUDE_DIR because cmake fails to find it otherwise.
+# Same is true for the boost python libs.
+setup(
+    name="pyopenvdb",
+    use_scm_version={
+        "root": '../../..',
+        "relative_to": __file__,
+    },
+    author="OpenVDB",
+    author_email="openvdb-dev@lists.aswf.io",
+    description=(
+        "Python bindings for OpenVDB: sparse volume data structure and tools."
+    ),
+    download_url="https://pypi.python.org/pypi/pyopenvdb",
+    headers=['pyopenvdb.h'],
+    include_package_data=True,
+    install_requires=[
+        'cmake>=3.15.3',
+        'numpy>=1.17.0'
+    ],
+    license="Mozilla Public License 2.0",
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    url="https://github.com/AcademySoftwareFoundation/openvdb",
+    cmake_args=[
+        '-DUSE_NUMPY=ON',
+        '-DOPENVDB_PYTHON_WRAP_ALL_GRID_TYPES=ON',
+        '-DOPENVDB_BUILD_CORE=ON',
+        f'-DNUMPY_INCLUDE_DIR={np.get_include()}',
+        f'-DBoost_PYTHON_VERSION={sys.version_info[0]}.{sys.version_info[1]}',
```

Comment:
Doe this build from the root of OpenVDB or from openvdb/openvdb/python? Our CMake files aren't really setup to be built from any sub-directory so I'm surprised this works if it's the latter?

Reply:
This builds from `openvdb/openvdb/python`. In fact, when the `sdist` is downloaded by a user from PYPI, they only get the contents of `openvdb/openvdb/python`. It's convenient that this can successfully build (for now), but it also sounds like there's potentially some pitfalls here if we don't build from the root of the repo...?

Reply:
How is it finding the core library target? In normal cmake builds, the python module will either use the generated `openvdb` target or an existing `OpenVDB::openvdb` target which is imported from the system:

```
if(NOT OPENVDB_BUILD_CORE)
  set(OPENVDB_LIB OpenVDB::openvdb)  # we're not building vdb - this target is defined by the rot cmake which tries to look for vdb on the system
else()
  set(OPENVDB_LIB openvdb) # we're building the core lib so link and use that
endif()
```

But these targets are defined outside of the CMake in `openvdb/openvdb/python` -

Reply:
When users install this, they won't be getting the whole repository - just the python bindings. When the user builds the sdist during installation, it will need to link to the system version of openvdb - just as `numpy` does with `LAPACK`.